### PR TITLE
add workflow setup for native example app

### PR DESCRIPTION
**Pull request by josemak25**
### Summary
This PR adds config for CI pipleline

### CODE REVIEW

File: reassure-tests.sh
Comment (mdjastrzebski) [inline reassure-tests.sh:1]: There should be no top-level `reassure-tests.sh`. Each example app should have its own `reassure-tests.sh` file in the their root folder.

File: package.json [hunk 1]
```diff
@@ -20,5 +20,8 @@
     "nohoist": [
       "examples/**"
     ]
+  },
+  "dependencies": {
```
Comment (mdjastrzebski) [inline package.json:24]: There is no need to add `reassure` dependency in top-level package.json. It's a dummy file just to make Dependabot be able to notice all relevant `examples/*/package.json` files.

Comment (josemak25) [reply package.json:24]: This means I should remove the reassure dependency right?

File: yarn.lock
Comment (mdjastrzebski) [inline yarn.lock:1]: Pls remove this file.

File: .github/workflows/test-example-web-vite.yml
Comment (mdjastrzebski) [inline .github/workflows/test-example-web-vite.yml:1]: Pls remove workflow files for all examples except `native`.

File: .github/workflows/test-example-native.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14.x
+          cache: 'yarn'
+
+      - name: Validate TypeScript
+        run: yarn typecheck
```
Comment (mdjastrzebski) [inline .github/workflows/test-example-native.yml:20]: Use `yarn --cwd examples/native [command]` for all commands to run them in the proper folder.

File: .github/workflows/test-example-native.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+      - name: Validate ESLint
+        run: yarn lint
+
+      - name: Run tests
+        run: yarn test
+
+      - name: Install and build
+        run: cd examples/native && yarn install
```
Comment (mdjastrzebski) [inline .github/workflows/test-example-native.yml:29]:
```suggestion
        run: yarn --cwd examples/native install
```

File: .github/workflows/test-example-native.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Validate TypeScript
+        run: yarn typecheck
+
+      - name: Validate ESLint
+        run: yarn lint
+
+      - name: Run tests
+        run: yarn test
+
+      - name: Install and build
```
Comment (mdjastrzebski) [inline .github/workflows/test-example-native.yml:28]: Yarn install should be run before typecheck, eslint, tests, etc as they rely on NPM packages being already installed.

**Review by mdjastrzebski — CHANGES_REQUESTED** [review]

File: yarn.lock
Comment (josemak25) [reply yarn.lock:1]: Noted

File: dangerfile.js
[the remark this answers is not in the document]
Comment (josemak25) [reply dangerfile.js:1]: Noted

File: .github/workflows/test-example-web-vite.yml
Comment (josemak25) [reply .github/workflows/test-example-web-vite.yml:1]: Alright

**Review by mdjastrzebski — APPROVED** [review]

**Pull request merged**